app/src/main/java/com/android/sample/ui/createRecipe/OptionalInformationScreen.kt

```diff
@@ -175,3 +178,20 @@ private fun SubtitleWithDropDown(
           Modifier.fillMaxWidth().padding(horizontal = PADDING_16.dp).testTag(dropdownTestTag),
       defaultItemIndex = defaultItemIndex)
 }
+
+/**
+ * Helper function to navigate to the next screen based on the isEditing flag.
+ *
+ * @param isEditing Boolean indicating whether the recipe is being edited.
+ * @param navigationActions Actions for navigating between screens.
+ */
+fun fromOptionalInformationScreenNavigateToNextScreen(
+    isEditing: Boolean,
+    navigationActions: NavigationActions
+) {
+  if (isEditing) {
+    navigationActions.navigateTo(Screen.EDIT_RECIPE_LIST_INGREDIENTS)
+  } else {
+    navigationActions.navigateTo(Screen.CREATE_RECIPE_INGREDIENTS)
+  }
+}
```

Comment:
I did this because of cognitive complexety issue raised by sonarcloud. What you pointed out was my previous implmentation. So I dont think that is a good idea to come back to this implementation as it doesnt improve anything anyway